add feature Search engine

mahdi00021 opened the pull request:
please say to me that I did the change right?

andersevenrud commented on src/desktop.js:
```diff
@@ -130,8 +130,8 @@ export default class Desktop extends EventEmitter {
      */
-    this.search = core.config('search.enabled') ? new Search(core) : null;
-
+    //this.search = core.config('search.enabled') ? new Search(core) : null;
```
Don't include commented code.

mahdi00021 replied:
> Don't include commented code.

You were supposed to tell me where I changed the wrong code and I will go and correct it

andersevenrud commented on src/search.js:
```diff
@@ -39,7 +39,13 @@ export default class Search {
+  
+  
+    const providedAdapters = options.adapters || [];
+    const useAdapters = [VFSSearchAdapter, ...providedAdapters];
```
The `VFSSearchAdapter` adapter needs to be written.

mahdi00021 replied:
@andersevenrud 
you said that only MySearchAdapter should be define not VFSSearchAdapter

andersevenrud replied:
This is why I said that my code was only for illustration purposes.

There was more changes needed, but I didn't write it because I explained what was needed.

andersevenrud commented on src/search.js:
```diff
@@ -119,8 +127,13 @@ export default class Search {
           });
       });
 
```
Don't include commented code.

andersevenrud reviewed (changes requested):
It looks like you copy/pasted my example code with comments and everything. I wrote that for illustrative purposes, and not to be used as-is :)

So there are a couple of problems here that I have commented on.

The Github Actions also have detected formatting issues with your code.

mahdi00021 commented:
> 

You said to change it, I changed it, this is not the final version

I said that I am new to the project and am not fluent in it, so it is not right to make fun of it

I changed the code according to your words, where is the funny part?

mahdi00021 commented:
@andersevenrud 
You were supposed to tell me where I changed the wrong code and I will go and correct it

andersevenrud commented:
I don't understand... I am not making fun of anything.


On Sun, 11 Dec 2022, 18:19 mahdi norouzi, ***@***.***> wrote:

> You said to change it, I changed it, this is not the final version
>
> I said that I am new to the project and am not fluent in it, so it is not
> right to make fun of it
>
> I changed the code according to your words, where is the funny part?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/os-js/osjs-client/pull/199#issuecomment-1345609883>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABHODBO6TADM2FJRV2IG7TWMYELRANCNFSM6AAAAAASZM4W2A>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

mahdi00021 commented:
Since I am new, please explain more about the changes to the codes
